packages/commonwealth/client/scripts/views/components/SublayoutHeader/DesktopHeader/DesktopHeader.tsx

```diff
@@ -98,6 +101,21 @@ const DesktopHeader = ({ onMobile, onAuthModalOpen }: DesktopHeaderProps) => {
             />
 
             <HelpMenuPopover />
+            <CWDropdown
+              containerClassName="language-selector"
+              label={`${SUPPORTED_LANGUAGES[languageStore.getState().currentLanguage].flag}`}
+              options={Object.entries(SUPPORTED_LANGUAGES).map(
+                ([code, { name, flag }]) => ({
+                  label: `${flag} ${name}`,
+                  value: code,
+                }),
+              )}
+              onSelect={(item) =>
+                languageStore
+                  .getState()
+                  .setLanguage(item.value as keyof typeof SUPPORTED_LANGUAGES)
+              }
+            />
```

Comment:
this should be behind a feature flag, as it doesn't actually change the app language yet.